optimum/exporters/tasks.py

```diff
@@ -665,38 +668,43 @@ def determine_framework(model: str, framework: str = None) -> str:
             logger.info(f"Local {framework_map[framework]} model found.")
         else:
             try:
-                AutoModel.from_pretrained(model)
+                url = huggingface_hub.hf_hub_url(model_name, WEIGHTS_NAME, subfolder=subfolder)
+                huggingface_hub.get_hf_file_metadata(url)
                 framework = "pt"
             except Exception:
                 pass
 
             if framework is None:
                 try:
-                    TFAutoModel.from_pretrained(model)
+                    url = huggingface_hub.hf_hub_url(model_name, TF2_WEIGHTS_NAME, subfolder=subfolder)
+                    huggingface_hub.get_hf_file_metadata(url)
                     framework = "tf"
                 except Exception:
                     pass
 
-            if framework is None:
-                if is_torch_available():
-                    framework = "pt"
-                elif is_tf_available():
-                    framework = "tf"
-                else:
-                    raise EnvironmentError("Neither PyTorch nor TensorFlow found in environment. Cannot export model.")
+        if is_torch_available():
```

Comment:
what if `framework == "pt"` and pytorch is not available? `framework == "tf"` and tensorflow not available? I'm not sure to understand this last logic

Reply:
It is handled just after:
```python
        if is_torch_available():
            framework = framework or "pt"
        elif is_tf_available():
            framework = framework or "tf"
        else:
            raise EnvironmentError("Neither PyTorch nor TensorFlow found in environment. Cannot export model.")
```